### DISCUSSION TITLE
Restrict PdfViewerParams.layoutPages to optimize _guessCurrentPage and _customPaint

### DISCUSSION THREAD

**Discussion by Dunrar** [espresso3389/pdfrx#240] [Ideas]
Hi, great project! My question is if you think it might be worth it to restrict `PdfViewerParams.layoutPages` in some way to more efficiently check page intersection with the currently visible rect (or cacheExtentRect) via binary search. Or is the intersection computation for all pages negligible, even though it runs so often?

**Comment by espresso3389** [comment]
You mean, you have performance issues with it?

**Reply by Dunrar** [reply]
No, just an idea. I wrote a much inferior PDF viewer using PDFium and Dart FFI a while ago and while comparing the codebases I saw that I used binary search to check visibility, but this was only possible because because my viewer can only lay out pages vertically or horizontally.

**Reply by espresso3389** [reply]
I see. You are correct to some extent. Frankly speaking, I've built the layout logic too much generic and it resulted in several uncontrollable issues.

But, for performance, I don't receive any such comments/complains so far.

**Reply by Dunrar** [reply]
Alright, no worries, I don't really have any performance complaints either. It was just something I thought I might be able to implement if restricting the layout is something you would want.
